src/librustc_mir/interpret/validity.rs

```diff
@@ -776,14 +781,14 @@ impl<'rt, 'mir, 'tcx, M: Machine<'mir, 'tcx>> ValueVisitor<'mir, 'tcx, M>
                         // For some errors we might be able to provide extra information
                         match err.kind {
                             err_ub!(InvalidUndefBytes(Some(ptr))) => {
-                                // Some byte was undefined, determine which
+                                // Some byte was uninitialized, determine which
                                 // element that byte belongs to so we can
                                 // provide an index.
                                 let i = usize::try_from(ptr.offset.bytes() / layout.size.bytes())
                                     .unwrap();
                                 self.path.push(PathElem::ArrayElem(i));
 
-                                throw_validation_failure!("undefined bytes", self.path)
+                                throw_validation_failure!("uninitialized value", self.path)
```

Comment:
I changed "bytes" to "value" because the thing we point at could have any integer of floating point type -- calling it a "byte" seems strange (and the plural is also not necessarily right here).

Reply:
I think "bytes" is correct, if you interpret is as the bytes of the memory holding the value, this is how I thought of it all the time to keep it in sync between indirect and immediate operands

Reply:
> this is how I thought of it all the time to keep it in sync between indirect and immediate operands

I don't know what you mean by this.

> I think "bytes" is correct, if you interpret is as the bytes of the memory holding the value

Hm yeah I think I can see that.

Reply:
> I don't know what you mean by this.

Well, indirect operands point to a bunch of bytes holding the value. Immediate operands are an optimization so we don't have to go through the memory all the time.

Reply:
Yes but what is the relation to the error?

Reply:
My confusion has been resolved. Let's use `uninitialized` everywhere and stop using `undefined`.